File: csharp/剑指39数组中出现次数超过一半的数字.cs

public partial class Solution
{
    private int _getMajority(int[] nums, int l, int r)
    {
        if (l == r)
            return nums[l];
        int lv, rv;
        lv = _getMajority(nums, l, (l + r) >> 1);
        rv = _getMajority(nums, ((l + r) >> 1) + 1, r);

        if(lv==rv)
            return lv;

        int lc=0, rc=0;
        for (int i = l; i<=r;i++)
        {
            if(nums[i]==lv)
                lc++;
            if(nums[i]==rv)
                rc++;
        }

        return lc > rc ? lv : rv;

    }
    // 分治
    public int MajorityElement(int[] nums)
    {
        return _getMajority(nums, 0, nums.Length - 1);
    }
}

File: python/main/剑指39数组中出现次数超过一半的数字.py
from collections import Counter

# 哈希
def majorityElement(nums: list) -> int:
    return Counter(nums).most_common(1)[0][0]

# 排序
def majorityElement1(nums: list) -> int:
    return sorted(nums)[len(nums)//2]

print(majorityElement1([3,1,2,3,3]))